File: book/스프링부트를 활용한 마이크로서비스 개발/test/src/main/java/com/example/test/service/MultiplicationService.java
package com.example.test.service;

import com.example.test.domain.Multiplication;
import com.example.test.domain.MultiplicationResultAttempt;

import java.util.List;

public interface MultiplicationService {
    Multiplication createRandomMultiplication();

    List<MultiplicationResultAttempt> getStatsForUser(String userAlias);

    boolean checkAttempt(final MultiplicationResultAttempt resultAttempt);

    MultiplicationResultAttempt getResultById(Long resultId);
}

File: book/스프링부트를 활용한 마이크로서비스 개발/kotlin/src/main/kotlin/com/example/kotlin/service/MultiplicationService.kt
package com.example.kotlin.service

import com.example.kotlin.domain.Multiplication
import com.example.kotlin.domain.MultiplicationResultAttempt


interface MultiplicationService {
    /*
    * 두 개의 무작위 인수를 인수를 담은 {@link Multiplication} 객체를 생성한다.
    * 무작위로 생성되는 숫자 범위는 11~99.
    *
    * @return 무작위 인수를 담은 {@link Multiplication} 객수
    * */
    fun createRandomMultiplication() : Multiplication

    /*
    * @return 곱셈 계산 결과가 맞으면 true, 아니면 false
    * */
    fun checkAttempt(resultAttempt: MultiplicationResultAttempt): Boolean
    fun getStatsForUser(userAlias: String): List<MultiplicationResultAttempt>
}